### DISCUSSION TITLE
Can you fix the position of frequency for dynamic feature to the band?

### DISCUSSION THREAD

**Discussion by acc4github** [ZL-Audio/ZLEqualizer#52] [Ideas]
The title says it all. It's a bit overly complex to use it sometimes that it's seperated.

**Comment by zsliu98** [comment]
Could you please explain which frequency are you talking about? If it is the centered frequency of the band-pass filter on the side-chain, it is a fixed value because people may want to trigger the compression(or expansion) from a different frequency area.

**Reply by acc4github** [reply]
Yes. I want where the dynamic compression is triggered is fixed to the matching band when I grab&move it around, without being seperated from it.
It's cool that ZL supports it to be seperated tho. But I don't really need it most of the time and it's an unnecessary work to adjust its position to match the selected band again and again.

**Reply by zsliu98** [reply]
Ah, it sounds like a good feature. Perhaps I can add another button to control whether these two are linked. I may replace the side-chain button with this button and move the side-chain control to the Dynamic dropdown list at the top.
Let's move it to an issue.

**Reply by acc4github** [reply]
It might be just a personal preference, but I think it'd be good if the compression band just shares every controls(including the Q range) with the original band, except for the controls for the compression itself(attack, release, knee, etc). :> And make it that way by default, while making it able to seperate it by a button. The compression band's existence is initially linked to the original band anyway. 

But again, it's just a feedback. Feel free!

**Reply by zsliu98** [reply]
@acc4github This feature is included in the pre-release 0.3.0. However, you have to enable it by turning on `General` -> `Dyn Link`. When `Dyn Link` is on, any adjustment to freq/Q of any main filter will also modify freq/Q of the corresponding band-pass filter.

**Comment by mikelupe** [comment]
The separation of the trigger frequency is an elementary part of any good dynamic EQ. Dynamically EQ-ing in general is a rather "complex" task even for experienced audio engineers, so this feature only reflects the task :)

But a "link/lock to target frequency range" could indeed be a nice additional feature.

**Reply by mikelupe** [reply]
This is a nice feature - thank you.

Would the effort be moderate to make this a per-frequency-point setting instead of global? 

Background is: when globally switching mode e.g. to linked, the unlinked points jump to a linked mode as well, as soon as one changes the base frequency of that point. So this can be "dangerous" in losing the original separation.

**Reply by zsliu98** [reply]
That sounds reasonable. Let's see where the button should be placed ... But I will also keep the global setting as some people may want that.

**Reply by acc4github** [reply]
Thanks for keeping the global setting option.

**Reply by mikelupe** [reply]
Makes absolutely sense - would be best as a "Default dynamic mode (linked/unlinked)" setting for new pointsm so it wouldn't affect already configured points.

**Reply by zsliu98** [reply]
@acc4github @mikelupe Now everything should work as expected. You can enable/disable the link of a single band by clicking the link button right above the dragger (see the new plugin screenshot). It will be included in the formal release of `0.3.0` (hopefully, next week).

**Reply by mikelupe** [reply]
Confirmed. 

Maybe you could also switch the icon/mode to unlinked, after a manual drag/separating of the trigger band happens, as this would reflect the unlinked mode it afterward is.

**Reply by zsliu98** [reply]
That depends. Perhaps some people would prefer to keep the default as linked and switch to unlinked when doing some fine-tuning. BTW, I also want to keep the logic between parameters simple as Steinberg suggests against that.

**Reply by mikelupe** [reply]
I see, then maybe lock it for good. 

Right now when moving the triggerband while it is linked, it separates, but band mode still remains linked. This manifests in a surprising way when the frequency band is then move/changed again: the trigger band immediatly jumps back to the frequency band. It's a bit confusing. So basically: "separation" only should be possible, when band mode is set to unlinked. Does that make sense?
 
edit: typo

**Reply by zsliu98** [reply]
The reason is I can only choose one direction link, either main filter -> trigger filter or trigger filter -> main filter. Otherwise, it may run into an endless loop. ~~However, I can disable the link automatically once you move the trigger filter manually. I will add this feature. I can't, there is another loop 😓~~ Maybe I can .. let me look into this

**Reply by mikelupe** [reply]
Oh, too bad. Well, it is what it is.

edit: your las suggestion was what I meant previously ("disable link once trigger filter manually moved)"

**Reply by zsliu98** [reply]
Now the link will be disabled once you click the dragger of the trigger band. However, it you adjust it from the bottom panel knobs, the link will stay the same.

**Comment by mikelupe** [comment]
Great doing on your part. Same for the Q/mouse issue.
